### PR TITLE
Revive SQLancer

### PR DESCRIPTION
**Pull request by qoega**
### Changelog category (leave one):
- Build/Testing/Packaging Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Run SQLancer for each pull request and commit to master. [SQLancer](https://github.com/sqlancer/sqlancer) is an OpenSource fuzzer that focuses on automatic detection of logical bugs.

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

File: tests/ci/sqlancer_check.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+            os.path.join(workspace_path, "status.txt"), "r", encoding="utf-8"
+        ) as status_f:
+            status = status_f.readline().rstrip("\n")
+
+        with open(
+            os.path.join(workspace_path, "summary.tsv"), "r", encoding="utf-8"
+        ) as summary_f:
+            for line in summary_f:
+                l=line.split('\t')
+                test_result.append((l[0],l[1]))
```
Comment (qoega) [inline tests/ci/sqlancer_check.py:157]:
```suggestion
                l = line.split('\t')
                test_result.append((l[0], l[1]))
```

File: tests/ci/sqlancer_check.py [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+        with open(
+            os.path.join(workspace_path, "status.txt"), "r", encoding="utf-8"
+        ) as status_f:
+            status = status_f.readline().rstrip("\n")
+
+        with open(
+            os.path.join(workspace_path, "summary.tsv"), "r", encoding="utf-8"
+        ) as summary_f:
+            for line in summary_f:
+                l = line.split('\t')
```
Comment (qoega) [inline tests/ci/sqlancer_check.py:156]:
```suggestion
                l = line.split("\t")
```

File: docker/test/sqlancer/run.sh [hunk 3]
```diff
@@ -1,33 +1,55 @@
...
+cd /workspace
+/clickhouse server -P /workspace/clickhouse-server.pid -L /workspace/clickhouse-server.log -E /workspace/clickhouse-server.log.err --daemon
+
+for _ in $(seq 1 30); do if [[ $(wget -q 'localhost:8123' -O-) == 'Ok.' ]]; then break ; else sleep 1; fi ; done
 
 cd /sqlancer/sqlancer-master
 
 export TIMEOUT=300
 export NUM_QUERIES=1000
+export NUM_THREADS=30
```
Comment (qoega) [inline docker/test/sqlancer/run.sh:40]:
```suggestion
export NUM_THREADS=20
```

**Review by qoega — COMMENTED** [review]
Just to restart CI.

**Review by rschu1ze — APPROVED** [review]

**Comment by azat** [comment]
Was it the intention that failures of SQLancer are simply ignored? (since it does not return status correctly)
For example in this commit https://github.com/ClickHouse/ClickHouse/commit/c12cfab953ef30cf90cc3ecaf0a8ce47f7a482ad the check is green, however if you will look at [the report](https://s3.amazonaws.com/clickhouse-test-reports/0/c12cfab953ef30cf90cc3ecaf0a8ce47f7a482ad/sqlancer__debug_.html) it fails

**Pull request merged**